manifests/install.pp

```diff
@@ -90,7 +89,7 @@
           "puppet:///modules/${caller_module_name}/${pack_filename}",
           "puppet:///files/packs/${pack_filename}"
         ],
-        require        => Wso2base::Clean_deployment['clean_on_pack_change'],
+        require        => [Wso2base::Ensure_directory_structures[$install_dirs], Wso2base::Clean_deployment['clean_on_pack_change']],
```

Comment:
Is there any reason to add following dependency?
(file_bucket) -> (Ensure_directory_structures[$install_dirs])

As I see we already have following dependency tree which includes Ensure_directory_structures at the last element:

(file_bucket) -> (Clean_deployment['clean_on_pack_change']) -> (Ensure_directory_structures[$install_dirs])

Reply:
The dependency between clean and directory creation is not needed IMO. Those should be done in that order, not the other way. The real relationship should be file copying depending on the existence of the directory structures.